services/provider/server/server.go

```diff
@@ -417,6 +425,51 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 
 	}
 
+	// Fetch noobaa remote secret and management address and append to extResources
+	clusterID := consumerResource.Status.Client.ClusterID
+	noobaaOperatorSecret := &v1.Secret{}
+	noobaaOperatorSecretName := fmt.Sprintf("noobaa-remote-join-secret-%s", clusterID)
+	err = s.client.Get(ctx, types.NamespacedName{Name: noobaaOperatorSecretName, Namespace: s.namespace}, noobaaOperatorSecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get %s secret. %v", noobaaOperatorSecretName, err)
+	}
+
+	authToken, ok := noobaaOperatorSecret.Data["auth_token"]
+	if !ok || len(authToken) == 0 {
```

Comment:
does this happen anytime that secret is created but data will not exist?

Reply:
Why does it matter? adding protection against this case will not harm us